tasks/component-usage-graph.js

```diff
@@ -0,0 +1,174 @@
+import fs from 'fs';
+import path from 'path';
+import chalk from 'chalk';
+import { DepGraph } from 'dependency-graph';
+import { init, parse } from 'es-module-lexer';
+import { getComponentFiles } from '../rollup/rollup-common.js';
+
+/**
+ *
+ * @returns {DepGraph} - returns a component `DepGraph`.
+ *
+ */
+export async function getComponentsGraph () {
+  const componentsGraph = new DepGraph();
+  const componentsFileList = getComponentFiles();
+  await init;
+
+  // Importer: One component that has one or more imports.
+  // Imported: One of the imports needed by the importer.
+  // e.g: cc-notice needs cc-icon. In this case, cc-notice is the importer and cc-icon the imported.
+
+  for (const importerFilePath of componentsFileList) {
+    const importerFile = fs.readFileSync(importerFilePath, 'utf8');
+    const [imports] = parse(importerFile);
+
+    for (const imported of imports) {
+
+      const importerRelative = path.relative(process.cwd(), importerFilePath);
+      const importerRelativeDir = path.dirname(importerRelative);
+      const sourceRelative = path.join(importerRelativeDir, imported.n);
+
+      if (!componentsGraph.hasNode(importerRelative)) {
+        componentsGraph.addNode(importerRelative);
+      }
+
+      const isImporterInList = componentsFileList.includes(importerRelative);
+      const isImportedInList = componentsFileList.includes(sourceRelative);
+      const isImportedLocal = imported.n.startsWith('./') || imported.n.startsWith('../');
```

Comment:
**question:** I'm confused by this line I don't understand why we need to check this :thinking: (it's not a request for change, just a request for help :sweat_smile:)

Reply:
We have to do this check because packages are dependencies thus not within our source files.
Here's an example: every component need `Lit` so they will have that line `import { css, html, LitElement } from 'lit';`. 
From the import we know that as there's no sign of a path here it's a dependency and we don't have to take this into account. I hope it's clearer for you that way :pray:

Reply:
Ah yes I gues it now, perfect explanation! Looks good to me :+1: